### PR TITLE
Add test for conflicting resource ref and labels

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation

https://issues.redhat.com/browse/APPSVC-248

###
testing:
`make test-unit`
or
`go test ./pkg/controller/servicebindingrequest -run TestReconcilerReconcileWithConflictingAppSelc -v `

### CODE REVIEW

**Comment by sbose78** [comment]
> Skipping this test due to an unresolved issue #319

So, this test wouldn't actually run ?

**Comment by pratikjagrut** [comment]
> > Skipping this test due to an unresolved issue #319
> 
> So, this test wouldn't actually run ?

Yes, it will not, until #319 is fixed.

**Comment by sbose78** [comment]
Could you fix #319 ?

**Comment by sbose78** [comment]
( ask @baijum for assistance if needed..)

**Comment by pratikjagrut** [comment]
> Could you fix #319 ?

I'm digging into it to find the crux of this issue, but I might gonna need help.

**Comment by pratikjagrut** [comment]
Ok, cool, I'll get in touch with @baijum.

**Comment by sbose78** [comment]
How's this going?

**Comment by pratikjagrut** [comment]
@sbose78 Looks like search() of binder.go is the crux.
[binder.go->L76](https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/binder.go#L76)
The list is returning everything for the given GVR even if options only contains resourceRef.

**Comment by sbose78** [comment]
Can we fix it ?

**Comment by pratikjagrut** [comment]
> Can we fix it ?

I'm working on it?

**Comment by pratikjagrut** [comment]
@sbose78 To close the story we can merge this PR, and once #319 is fixed I'll send another PR to remove the skip. I'm not sure how much time is going to take that bug to fix. As @baijum is busy with something I asked @akashshinde to help me with this.

**Comment by pratikjagrut** [comment]
@sbose78 #319 is not fixed yet but I found a workaround now this test works fine, this PR is oK to merge.

File: pkg/controller/servicebindingrequest/reconciler_test.go [hunk 1]
```diff
@@ -184,3 +184,43 @@ func TestReconcilerReconcileUsingVolumes(t *testing.T) {
...
+
+		res, err := reconciler.Reconcile(reconcileRequest())
+		require.NoError(t, err)
+		require.False(t, res.Requeue)
+
+		namespacedName := types.NamespacedName{Namespace: reconcilerNs, Name: reconcilerName}
+		sbrOutput, err := reconciler.getServiceBindingRequest(namespacedName)
+		require.NoError(t, err)
+
+		require.Equal(t, BindingSuccess, sbrOutput.Status.BindingStatus)
```
Comment (DhritiShikhar) [inline pkg/controller/servicebindingrequest/reconciler_test.go]: Can you add a check for `sbrOutput.Status.Conditions` as well?

File: pkg/controller/servicebindingrequest/reconciler_test.go [hunk 2]
```diff
@@ -184,3 +184,43 @@ func TestReconcilerReconcileUsingVolumes(t *testing.T) {
...
+		expectedStatus := fmt.Sprintf("%s/%s", reconcilerNs, applicationResourceRef2)
+		require.Equal(t, expectedStatus, sbrOutput.Status.ApplicationObjects[0])
```
Comment (DhritiShikhar) [inline pkg/controller/servicebindingrequest/reconciler_test.go]: Can you add a check that length of `sbrOutput.Status.ApplicationObjects` is 1 and there are no other applicationObjects?

File: pkg/controller/servicebindingrequest/reconciler_test.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/controller/servicebindingrequest/reconciler_test.go]: If we have `BindingSuccess` asserted then is there any use of asserting conditions?

Comment (DhritiShikhar) [reply pkg/controller/servicebindingrequest/reconciler_test.go]: @pratikjagrut We want to support conditions and deprecate BindingStatus in future

**Comment by Avni-Sharma** [comment]
@pratikjagrut  can you add a case like this one https://github.com/redhat-developer/service-binding-operator/commit/c0f53c17f7bbb85b8bae6df48b1147643165e385#diff-14a37dadc390fc3ec23459efc088de8eR256 . Apart from this, it looks good to me

**Comment by Avni-Sharma** [comment]
Cleanup of tests will happen in another PR as mentioned by @pratikjagrut . Merging this

File: pkg/controller/servicebindingrequest/reconciler_test.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/controller/servicebindingrequest/reconciler_test.go]: I added condition assertion.

**Comment by pratikjagrut** [comment]
> Cleanup of tests will happen in another PR as mentioned by @pratikjagrut . Merging this

I removed mock CSV, so no other PR is needed.

**Pull request merged**